### DISCUSSION TITLE
add support for onResize event / expose ref to list wrapper

### DISCUSSION THREAD

**Discussion by vilgeforc5** [inokawa/virtua#582] [Ideas]
Hi, we're using VList  for virtualized message list and we've encountered problem when message font size changes (after messages are mounted and everything)
The problem is that scroll isn't preserved for the very bottom, it kind of "jumps" up for a little bit. 
Currently, we manually use ResizeObserver for inner wrapper (getting it via el.firstElementChild?.firstElementChild?.firstElementChild)

I think it'll be good to add callback/expose ref so that such events can be handled "out of box"

**Comment by inokawa** [comment]
Similar feature was requested several times (#301, #408).

I'm basically positive to implement it but I don't have enough time to define good API. If you need it early, it would be helpful if you could create a PR.

**Comment by mariia-mitiureva** [comment]
@vilgeforc5 I use the same workaround (ResizeObserver for inner wrapper), and I couldn't make it scroll to bottom before the next repaint, so it still looks "jumpy". Do you have the same problem or does it work good for you?

**Reply by vilgeforc5** [reply]
It seems like update of item list and resize happens at the same time in your case (that's expected behavior)
you can try to use scrollToBottom({smooth: true}) and detect that you are at the very bottom of list (and defer/ignore resize update in that case)

**Reply by mariia-mitiureva** [reply]
The problem is that I can't make scroll "stick" to bottom. After resizing it is no longer at the bottom and after a moment it's scrolled back to bottom. It's noticeable

**Comment by mariia-mitiureva** [comment]
@inokawa  `onResize` is also needed to implement custom `onRangeChange`. `onScroll` is not enough to cover all cases
